Password optional: Sign in with email using JS React Guide (#3557)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/pwd-optional-sign-in-email/main/react/integrationsteps.md

### 1: The user navigates to the sign-in page

When the user navigates to your app's sign-in page, call `OktaAuth.idx.start()` to start the sign-in flow.

```javascript
const { oktaAuth } = useOktaAuth();
const startFlow = useCallback(async () => {
  newTransaction = await oktaAuth.idx.start();
}, [oktaAuth, flow, setTransaction]);
```

The method returns an `IdxTransaction` object containing field metadata that can be used to create a sign-in page dynamically.

```json
{
   "status":"PENDING",
   "nextStep":{
      "name":"identify",
      "inputs":[
         {
            "name": "username",
            "label": "Username",
            "required": true
         }
         ...
      ]
   },
}
```

Display &mdash; either statically or dynamically using `IdxTransaction` &mdash; an input field for the user's username.

<div class="half border">

![Screenshot showing a sign-in page with a username field and a Next button.](/img/pwd-optional/pwd-optional-sign-in-page.png)

</div>

### 2: The user submits their username

> When the user submits their account details, create an object with a `username` property and assign it the value entered by the user.

Reviewer: Is it worth saying here that their username is also their email address?
```suggestion
When the user submits their username, create an object with a `username` property and assign it the value entered by the user.
```

Author: Updated to username. Confusingly, when using the Embedded SDK, on signup the username is the username and primary email address.  During sign in the username doesn't have to be the primary email address since the account could have been created not using the Embedded SDK (in the case of using the Admin console).

```javascript
const inputValues = {
  username: "johndoe@email.com",
 };
```

> Send this new object to `OktaAuth.idx.proceed()`.

Reviewer: Sending a value to a method doesn't make sense.

```suggestion
Call `OktaAuth.idx.proceed()` passing in this new object as a parameter.
```

Author: Updated. But curiously, why doesn't "send ..." make sense?

```javascript
setProcessing(true);
const newTransaction = await oktaAuth.idx.proceed(inputValues);
setTransaction(newTransaction);
```

> >**Note**: You can also start the sign-in flow in a single step by passing the username in `OktaAuth.idx.authenticate()`. See [`idx.authenticate`](https://github.com/okta/okta-auth-js/blob/master/docs/idx.md#idxauthenticate) in the GitHub docs or [Node Express's](/docs/guides/pwd-optional-sign-in-email/nodeexpress/main/#_2-the-user-submits-their-username) version of this guide to learn more.

Reviewer: ```suggestion
>**Note**: You can also start the sign-in flow in a single step by passing the username as a parameter to `OktaAuth.idx.authenticate()`. See [`idx.authenticate`](https://github.com/okta/okta-auth-js/blob/master/docs/idx.md#idxauthenticate) in the GitHub docs or the [Node Express](/docs/guides/pwd-optional-sign-in-email/nodeexpress/main/#_2-the-user-submits-their-username) version of this guide to learn more.
```

Author: I've updated but like to chat about it.

> ### 3. Identity Engine requests new email verification

Reviewer: Can we remove the word "new" from the chevron in step 3 please?  
It reads "OIE is requesting the verification of a new email" rather than "OIE is requesting a new verification via an old email" which is what is happening as this is a sign-in flow rather than a sign-in flow or a change-your-password flow. Perhaps

3. Identity Engine requests sends user email challenge

Author: done.

> Identity Engine sends the user an email that contains a one-time password (OTP) they can use to verify their identity.`OktaAuth.idx.proceed()` returns an `IdxTransaction` object with a `status` of `PENDING`, indicating that the user needs to verify their identity with their email.

Reviewer: ```suggestion
Identity Engine sends the user an email that contains a one-time password (OTP) they can use to verify their identity. `OktaAuth.idx.proceed()` returns an `IdxTransaction` object with a `status` of `PENDING`, indicating that the user needs to verify their identity with their email.
```

Author: updated.

```json
{
  status: "PENDING",
  nextStep: {
    name: "enroll-authenticator",
    type: "email",
    authenticator: {
      type: "email",
      key: "okta_email",
      displayName: "Email",
    },
  },
}
```

Build the logic that handles this response and sends the user to a dialog where they enter the OTP.

### 4. The user verifies their identity with the new email

The user opens the email sent by Identity Engine. Create a dialog in your app where the user can submit the OTP from the email to Identity Engine.

<div class="half border">

![Screenshot showing a page with an OTP input field.](/img/pwd-optional/pwd-optional-sign-up-js-react-email-verify.png)

</div>

When the user submits the OTP, create an object with a `verificationCode` property set to the OTP entered by the user.

```json
const inputValues = {
    "verificationCode":"197277"
 };
```

> Call `OktaAuth.idx.proceed()` passing in the new object.

Reviewer: ```suggestion
Call `OktaAuth.idx.proceed()` passing in the new object as a parameter.
```

Author: Updated. Do we really need to say this?
